app/services/api/v3/ecf/unfunded_mentors_query.rb

```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
+
+module Api
+  module V3
+    module ECF
+      class UnfundedMentorsQuery
+        def initialize(lead_provider:, params:)
+          @lead_provider = lead_provider
+          @params = params
+        end
+
+        def unfunded_mentors
+          join = InductionRecord
+                   .select("DISTINCT FIRST_VALUE(induction_records.id) OVER (#{latest_induction_record_order}) AS latest_id")
+                   .joins(:participant_profile, induction_programme: :partnership)
+                   .where(
+                     induction_programme: {
+                       partnerships: {
+                         challenged_at: nil,
+                         challenge_reason: nil,
+                       },
+                     },
+                   )
+
+          funded_mentors = InductionRecord
+                   .select("induction_records.id, induction_records.mentor_profile_id")
+                   .joins(join_mentor_profiles)
+                   .joins(join_mentor_participant_identities)
+                   .joins(join_induction_programme)
+                   .joins("JOIN (#{join.to_sql}) AS latest_induction_records ON latest_induction_records.latest_id = induction_records.id")
+
+          unfunded_mentors = InductionRecord
+                   .select("induction_records.id, induction_records.participant_profile_id, partnerships.lead_provider_id, induction_records.created_at, induction_records.updated_at, induction_records.preferred_identity_id")
+                   .joins("JOIN participant_profiles ON participant_profiles.id = induction_records.participant_profile_id")
+                   .joins("LEFT OUTER JOIN (#{funded_mentors.to_sql}) AS funded_mentors_induction_records ON funded_mentors_induction_records.id = induction_records.id")
+                   .joins("JOIN (#{join.to_sql}) AS latest_induction_records ON latest_induction_records.latest_id = induction_records.id")
+                   .joins("JOIN induction_programmes on induction_records.induction_programme_id = induction_programmes.id")
+                   .joins("JOIN partnerships on partnerships.id = induction_programmes.partnership_id")
+                   .where("funded_mentors_induction_records.id is null")
+                   .where("participant_profiles.type = 'ParticipantProfile::Mentor'")
+
+          scope = User
```

Comment:
I think the query might be returning more than we need, let's catchup as easier to chat through, also we will need to amend the specs